client/app/bundles/course/assessment/submission/containers/QuestionGrade.jsx

```diff
@@ -7,25 +7,61 @@ import PropTypes from 'prop-types';
 import actionTypes from '../constants';
 import { questionGradeShape, questionShape } from '../propTypes';
 
+const GRADE_STEP = 1;
+
 const styles = {
   container: {
     marginTop: 20,
   },
 };
 
+/**
+ * Checks if the given value is a valid decimal of the form `0.00`.
+ *
+ * @param {string} value the string to be checked
+ * @returns `true` if `value` is a valid decimal
+ */
+const isValidDecimal = (value) => /^\d*(\.\d?)?$/.test(value);
+
 class VisibleQuestionGrade extends Component {
-  handleGradingField(value) {
+  handleFieldChange(value) {
```

Comment:
Do we need to have the return statements when `onChange` and `onBlur` are not expecting any return? Also, you might want to extract the common checks in both `handleFieldChange` and `handleFieldBlur`.